lib/takeaway.rb

```diff
@@ -0,0 +1,67 @@
+require_relative "menu"
+require_relative "sms"
+
+class Takeaway
+
+  def initialize(menu: Menu.new,sms: SMS.new)
+    @menu_class = menu
+    @sms = sms
+    @selected_dishes = Hash.new(0)
+  end
+
+  def list_dishes
+    @menu_class.dishes
+  end
+
+  def order
+    while true
+      numbered_list_dishes
+      input = user_input
+      break if input == 99
+      dish = @menu_class.dishes.keys[input - 1]
+      @selected_dishes[dish] += 1
+    end
+  end
+
+  def check_order
+    array = []
+    @selected_dishes.each do |key,value|
+      array << "#{key} x#{value} - £#{@menu_class.dishes[key] * value}"
+    end
+    array << "Total - £#{order_total}"
+    array.join("\n")
+  end
+
+  def complete_order
+    @sms.confirmation_text(confirmation_message)
+  end
+
+  private
+
+  def numbered_list_dishes
+    counter = 0
+    @menu_class.dishes.each do |key,value|
+      puts "#{counter + 1}. #{key} - £#{value}"
+      counter += 1
+    end
+    puts "99. Exit"
+  end
+
+  def user_input
```

Comment:
I really appreciate how interactive it is. And just how much it actually works like an app!